airbyte-lib/airbyte_lib/executor.py

```diff
@@ -119,19 +153,22 @@ def ensure_installation(self):
 
         connector_path = self._get_connector_path()
         if not connector_path.exists():
-            raise Exception(f"Could not find connector {self.metadata.name} in venv {venv_name}")
-
-        installed_version = self._get_installed_version()
-        if installed_version != self.target_version:
-            # If the version doesn't match, reinstall
-            self.install()
-
-            # Check the version again
-            version_after_install = self._get_installed_version()
-            if version_after_install != self.target_version:
-                raise Exception(
-                    f"Failed to install connector {self.metadata.name} version {self.target_version}. Installed version is {version_after_install}"
-                )
+            raise FileNotFoundError(
+                f"Could not find connector '{self.metadata.name}' " f"in venv '{venv_name}' with connector path '{connector_path}'."
+            )
+
+        if self.enforce_version:
+            installed_version = self._get_installed_version()
+            if installed_version != self.target_version:
+                # If the version doesn't match, reinstall
+                self.install()
```

Comment:
Only a suggestion, but I seem to run into a lot of python cases where I get '0.0.0' back - such as when installing from a local path. This might not be necessary, but I think it's worth considering. We can always implement as you've written and just keep an eye out for edge cases. It may already be solved in the latest handling (ignoring when a specific version is not requested), but I want to mention anyway while I'm here.

```suggestion
            if installed_version == '0.0.0':
                # pass or warn; version is either not declared or not detectable
                pass
            elif installed_version != self.target_version:
                # If the version doesn't match, reinstall
                self.install()
```

Reply:
I considered this in the interim, but I'm hoping that we don't need to hardcode this with the current scheme - if you are working with local / pip-url installed connectors you would just not worry about the version (and thus not specify it) and airbyte-lib won't worry either, so it won't hit this.

I'll go ahead and merge and we can add the 0.0.0 exception if it keeps coming up in some scenarios.

Reply:
Makes sense! Thanks!